core/src/main/scala/org/bitcoins/core/psbt/PSBTMap.scala

```diff
@@ -563,18 +587,17 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
     if (isFinalized) {
       this
     } else {
+      require(
+        this.witnessUTXOOpt.isEmpty || this.nonWitnessOrUnknownUTXOOpt.isEmpty)
```

Comment:
move this up to the top (still holds in the finalized case and should really be an invariant of InputPSBTMap), also error message

Reply:
What I meant was that what this actually ensures is that they aren't both populated and that this should go all the way up to the top of `InputPSBTMap`